cmd/kubelet/app/server.go

```diff
@@ -966,9 +966,13 @@ func RunDockershim(c *componentconfig.KubeletConfiguration, dockershimRootDir st
 		SupportedPortForwardProtocols:   streaming.DefaultConfig.SupportedPortForwardProtocols,
 	}
 
+	cadvisorInterface, err := cadvisor.New(uint(c.CAdvisorPort), "docker", c.RootDirectory)
```

Comment:
I know you never call start on this, but I still don't like the idea of having a second cAdvisor manager in the Kubelet (at the very least, the port should be 0). Since you're only using a single method of the interface, I think it would be better to try and refactor out the piece you need here. I think you should be able to pull out a single static function?

/cc @dashpole